<launch>
  <arg name="MACHINE" default="localhost" />
  <arg name="keepout" default="false" />
  <arg name="launch_rviz" default="true" />
  <arg name="launch_map_server" default="true" />
  <arg name="use_machine" default="true" />
  <arg name="use_pictogram" default="true" />

  <machine name="localhost" address="localhost" if="$(arg use_machine)" />

  <!--
  <node name="publish_spot_hongo" pkg="roseus" type="roseus"
        args="$(find jsk_maps)/tools/publish_campus.l"
        machine="$(arg MACHINE)"
        output="screen" />
  -->
  <node pkg="jsk_maps" type="static_map_publisher.py" name="static_map_publisher" output="screen">
      <remap from="/map" to="/static_map/map" />
      <remap from="/map_meta_data" to="/static_map/map_meta_data" />

      <rosparam>
          initial_latitude: 35.713537979438996
          initial_longitude: 139.762305151285
          zoom_level: 18
          map_size: 1500
          map_type: osm
          initial_frame: world
          static_map_frame: static_map_frame
      </rosparam>
  </node>

  <node pkg="jsk_maps" type="static_lon_lat_transform_publisher.py" name="$(anon tf_hongo_main_gate)">
      <rosparam>
          from_frame_id: world
          from_latitude: 35.713537979438996
          from_longitude: 139.762305151285
          to_frame_id: hongo_main_gate
          to_latitude: 35.712995049681204
          to_longitude: 139.75967053364684
          diff_angle_from_to: 0.0
      </rosparam>
  </node>

  <!--
  Eng. 2 Building Section
  -->
  <node pkg="jsk_maps" type="static_lon_lat_transform_publisher.py" name="$(anon eng2_tfA_raw)">
      <rosparam>
          from_frame_id: world
          from_latitude: 35.713537979438996
          from_longitude: 139.762305151285
          to_frame_id: eng2
          to_latitude: 35.71443022192967
          to_longitude: 139.7615547854118
          diff_angle_from_to: 0.2
      </rosparam>
  </node>

  <include file="$(find jsk_maps)/launch/include/start_map_eng2_main.xml">
    <arg name="prefix" value="eng2_" />
    <arg name="keepout" value="$(arg keepout)"/>
    <arg name="launch_map_server" value="$(arg launch_map_server)"/>
    <arg name="launch_map_select" value="false"/>
  </include>

  <node pkg="tf" type="static_transform_publisher" name="$(anon eng2_tfB)"
        args="0 6.9 0 0 0 0 /eng2/7f /eng2/7f/73B2 100"/>

  <node name="publish_spot_eng2" pkg="roseus" type="roseus"
        args="$(find jsk_maps)/tools/publish_spot.l"
        machine="$(arg MACHINE)"
        output="screen" >
    <param name="~scene" value="eng2" />
    <param name="~use_pictogram" value="$(arg use_pictogram)" />
  </node>

  <!--
  Eng. 8 Building Section
  -->
  <node pkg="jsk_maps" type="static_lon_lat_transform_publisher.py" name="$(anon eng8_tfA_raw)">
      <rosparam>
          from_frame_id: world
          from_latitude: 35.713537979438996
          from_longitude: 139.762305151285
          to_frame_id: eng8
          to_latitude: 35.71470007319424
          to_longitude: 139.760512685666
          diff_angle_from_to: 1.8
      </rosparam>
  </node>

  <include file="$(find jsk_maps)/launch/include/start_map_eng8_main.xml">
    <arg name="prefix" value="eng8_" />
    <arg name="keepout" value="$(arg keepout)"/>
    <arg name="launch_map_server" value="$(arg launch_map_server)"/>
    <arg name="launch_map_select" value="false"/>
  </include>

  <node pkg="tf" type="static_transform_publisher" name="$(anon eng8_tfB)"
        args="0 -13.678 0  0 0 0 /eng8/6f /eng8/6f/610 100" />

  <node name="publish_spot_eng8" pkg="roseus" type="roseus"
        args="$(find jsk_maps)/tools/publish_spot.l"
        machine="$(arg MACHINE)"
        output="screen" >
    <param name="~scene" value="eng8" />
    <param name="~use_pictogram" value="$(arg use_pictogram)" />
  </node>

  <!-- for initialpose topic -->
  <node name="initialpose3d" pkg="jsk_maps" type="initialpose3d.py">
    <remap from="initialpose_in" to="initialpose3d" />
    <remap from="initialpose_out" to="initialpose" />
  </node>

  <!-- mtf selector for navigation nodes -->
  <node name="tf_mux" pkg="topic_tools" type="mux"
        args="tf  /eng2/7f_tf /eng2/1f_tf /eng2/2f_tf /eng2/3f_tf /eng2/8f_tf /eng8/6f_tf /eng8/b1f_tf /eng8/1f_tf">
    <remap from="mux" to="map_tf_mux" />
  </node>
  <!-- rviz -->
  <node name="rviz" pkg="rviz" type="rviz"
        args="-d $(find jsk_maps)/config/hongo.rviz"
        if="$(arg launch_rviz)"/>
</launch>
